### PR TITLE
impl lite::Commit for commit::SignedHeader

### PR DESCRIPTION
**Pull request by yihuang**

### CODE REVIEW

File: tendermint/src/rpc/endpoint/commit.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //! `/commit` endpoint JSONRPC wrapper
 
-use crate::{block, rpc};
+use crate::{amino_types::vote::CanonicalVote, block, lite, rpc, vote::SignedVote, Hash};
```
Comment (Liamsi) [inline tendermint/src/rpc/endpoint/commit.rs]: The rpc module currently does not depend on `amino_types`. I think, if possible, we should try to keep it that way. Even if that means introducing another type  under tendermint/block (or similar).

**Comment by Liamsi** [comment]
Did the suggested changes here: https://github.com/interchainio/tendermint-rs/pull/63/commits/e1a7560e4dfbeaaa3f9013e2141b7296e33f2b5d

**Comment by yihuang** [comment]
Great, thanks.

File: tendermint/src/rpc/endpoint/commit.rs [hunk 2]
```diff
@@ -46,3 +46,26 @@ pub struct SignedHeader {
...
+impl lite::Commit for SignedHeader {
+    type Vote = SignedVote;
+    fn header_hash(&self) -> Hash {
+        self.commit.block_id.hash
+    }
+    fn into_vec(&self) -> Vec<Option<Self::Vote>> {
+        let chain_id = self.header.chain_id.to_string();
+        let mut votes = self.commit.precommits.clone().into_vec();
+        votes
+            .drain(..)
```
Comment (ebuchman) [inline tendermint/src/rpc/endpoint/commit.rs]: @yihuang Why use `drain` here rather than `iter` ?

Comment (yihuang) [reply tendermint/src/rpc/endpoint/commit.rs]: Yeah, my fault, not familiar enough with idiomatic rust then. But I believe this code already modified by @Liamsi in the final merged version ;D

**Pull request merged**